### ISSUE TITLE
Inconsistent results with EnsembleVoteClassifier

### ISSUE THREAD

**Issue by vascosa** [rasbt/mlxtend#712]
Hello,

I am combining a series of models with an EnsembleVoteClassifier:

`ensemble=EnsembleVoteClassifier(clfs=top_models, voting='soft',weights=None,fit_base_estimators=True,verbose=1)`

The top_models is a list of 5 classifiers, all with random states set:

```
[PipelineRFE(memory=None,
            steps=[('scaler', MinMaxScaler(copy=True, feature_range=(0, 1))),
                   ('ovl_coeff',
                    SelectKBest(k=207,
                                score_func=<function chi2 at 0x7f0d636e81f0>)),
                   ('classifier',
                    AdaBoostClassifier(algorithm='SAMME.R',
                                       base_estimator=SVC(C=1.0,
                                                          break_ties=False,
                                                          cache_size=200,
                                                          class_weight='balanced',
                                                          coef0=0.0,
                                                          decision_function_shape='ovr',
                                                          degree=3,
                                                          gamma='scale',
                                                          kernel='rbf',
                                                          max_iter=-1,
                                                          probability=True,
                                                          random_state=42,
                                                          shrinking=True,
                                                          tol=0.001,
                                                          verbose=False),
                                       learning_rate=1.2, n_estimators=10,
                                       random_state=42))],
            verbose=False), PipelineRFE(memory=None,
            steps=[('scaler', MinMaxScaler(copy=True, feature_range=(0, 1))),
                   ('kbest',
                    SelectKBest(k=312,
                                score_func=<function f_classif at 0x7f0d636dde50>)),
                   ('classifier',
                    LogisticRegression(C=4.0, class_weight='balanced',
                                       dual=False, fit_intercept=True,
                                       intercept_scaling=1, l1_ratio=None,
                                       max_iter=100, multi_class='auto',
                                       n_jobs=None, penalty='l1',
                                       random_state=42, solver='saga',
                                       tol=0.0001, verbose=0,
                                       warm_start=False))],
            verbose=False), PipelineRFE(memory=None,
            steps=[('scaler', MinMaxScaler(copy=True, feature_range=(0, 1))),
                   ('ovl_coeff',
                    SelectKBest(k=345,
                                score_func=<function mutual_info_classif at 0x7f0d636f91f0>)),
                   ('classifier',
                    LogisticRegression(C=1.0, class_weight='balanced',
                                       dual=False, fit_intercept=True,
                                       intercept_scaling=1, l1_ratio=None,
                                       max_iter=100, multi_class='auto',
                                       n_jobs=None, penalty='l2',
                                       random_state=42, solver='liblinear',
                                       tol=0.0001, verbose=0,
                                       warm_start=False))],
            verbose=False), PipelineRFE(memory=None,
            steps=[('scaler', MinMaxScaler(copy=True, feature_range=(0, 1))),
                   ('ovl_coeff',
                    SelectKBest(k=345,
                                score_func=<function mutual_info_classif at 0x7f0d636f91f0>)),
                   ('classifier',
                    SVC(C=4.0, break_ties=False, cache_size=200,
                        class_weight='balanced', coef0=0.0,
                        decision_function_shape='ovr', degree=3, gamma=0.01,
                        kernel='rbf', max_iter=-1, probability=True,
                        random_state=42, shrinking=True, tol=0.001,
                        verbose=False))],
            verbose=False), PipelineRFE(memory=None,
            steps=[('scaler', MinMaxScaler(copy=True, feature_range=(0, 1))),
                   ('kbest',
                    SelectKBest(k=624,
                                score_func=<function f_classif at 0x7f0d636dde50>)),
                   ('classifier',
                    ExtraTreesClassifier(bootstrap=True, ccp_alpha=0.0,
                                         class_weight='balanced',
                                         criterion='gini', max_depth=None,
                                         max_features=None, max_leaf_nodes=None,
                                         max_samples=0.6,
                                         min_impurity_decrease=0.0,
                                         min_impurity_split=None,
                                         min_samples_leaf=2,
                                         min_samples_split=2,
                                         min_weight_fraction_leaf=0.0,
                                         n_estimators=50, n_jobs=None,
                                         oob_score=True, random_state=42,
                                         verbose=0, warm_start=False))],
            verbose=False)]
```

When I predict the labels on a test set, the results are inconsistent, varying around +/-5% balanced accuracy. I have also imported the random module and set the random state seed there..

Is this normal behavior for MlXtend? If not, what am I doing wrong?

Thank you for you help,

Vasco

**Comment by rasbt** [comment]
Hi Vasco,

hm, this shouldn't happen (otherwise, the unit tests would also fail I suppose). Not sure where the randomness comes from, either (there is no randomness in the EnsembleVoteClassifier itself). I think 

> imported the random module and set the random state seed there..

usually doesn't do much for scikit-learn. You could try 

```python
import numpy as np
np.random.seed(42)    
```

though and see if that solves the issue.
